entities/weapons/weapon_ttt_basegrenade.lua

```diff
@@ -59,10 +61,12 @@ function SWEP:Throw()
 		e:SetWeapon(self)
 		e:Spawn()
 
-
+        self:TakePrimaryAmmo(1)
 		self:SetThrowStart(math.huge)
-		hook.Run("DropCurrentWeapon", self:GetOwner())
-		self:Remove()
+        if !(self:CanPrimaryAttack()) then
```

Comment:
this does not match our coding style (do `(!self:CanPrimaryAttack())`
you should also change this to check if current ammo is 0 instead of that anyways.

Reply:
Changed to check if current ammo 0